app/models/food_preference.rb

```diff
@@ -1,5 +1,10 @@
 class FoodPreference < ActiveRecord::Base
   has_many :dishes
+  after_commit :reindex_dishes
```

Comment:
There are 4 ActiveRecord callbacks that trigger a reindex with similar `reindex_dishes` methods on 4 model classes. 

It feels like a dedicated callback object is worth thinking about introducing. For an example see the `EncryptionWrapper` code at http://api.rubyonrails.org/classes/ActiveRecord/Callbacks.html